sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java

```diff
@@ -35,5 +39,20 @@
      */
     DefaultAzureCredential(ArrayDeque<TokenCredential> tokenCredentials) {
         super(tokenCredentials);
+        this.tokenCredentials = new ArrayList<TokenCredential>(tokenCredentials.size());
+        this.tokenCredentials.addAll(tokenCredentials);
+    }
+
+    /**
+     * Get the list of credentials sequentially used by {@link DefaultAzureCredential} to attempt authentication.
+     * Any changes made to the returned list will not reflect in the list of credentials
+     * used by {@link DefaultAzureCredential} to authenticate.
+     * The credentials in the returned list and their order may change in future versions of Identity.
+     * This API is not intended to be used in production ready code and should only be used for development purposes.
+     *
+     * @return The list of {@link TokenCredential}.
+     */
+    public List<TokenCredential> getCredentials() {
+        return tokenCredentials;
```

Comment:
This should return an unmodifiable list. Create the unmodifiable list in the ctor itself.

Reply:
made it unmodifiable.
If users want to modify it, they can create a modifiable list from it.